src/lando/main/scm/git.py

```diff
@@ -257,9 +257,17 @@ def describe_commit(self, revision_id: str = "HEAD") -> CommitData:
         """Return Commit metadata."""
         return self._describe_commits(revision_id)[0]
 
-    def describe_local_changes(self) -> list[CommitData]:
-        """Return a list of the Commits only present on this branch."""
-        return list(reversed(self._describe_commits("@{u}..")))
+    def describe_local_changes(
+        self, target_cset: str | None = None
+    ) -> list[CommitData]:
+        """Return a list of the Commits only present on this branch.
+
+        Use the passed target changeset as the base commit. Otherwise, use the
+        configured upstream branch.
+        """
+        refspec = f"{target_cset}.." if target_cset else "@{u}.."
```

Comment:
... for simplicity with the previous suggested change:

```suggestion
        refspec = f"{base_cset}.."
```